packages/core/src/providers/window/provider.tsx

```diff
@@ -0,0 +1,28 @@
+import { ReactNode, useState, useEffect } from 'react'
+import { WindowContext } from './context'
+
+interface Props {
+  children: ReactNode
+}
+
+/**
+ * @internal Intended for internal use - use it on your own risk
+ */
+export function WindowProvider({ children }: Props) {
+  const [isActiveWindow, setActiveWindow] = useState(true)
+
+  useEffect(() => {
+    document.addEventListener('visibilitychange', () => {
```

Comment:
We should add an event listener, and remove on unmounting.

So at the end of useEffect
```
return () => document.removeEventListener(...)
```